### PR TITLE
[Ml Data Frame] Size the GET stats search by number of Ids requested

### LINKED ISSUE #43203: [ML] If stopping _all data frames, then querying _all stats is not correct
Found in 7.2.0-BC5

1. Create and start 100 data frame transforms (same transform, incrementing id,  scripted, and can be on small source indices) - give them time to complete
2.  Check in UI - all transforms are visible in list and show 100% progress
3. Run `POST _data_frame/transforms/_all/_stop`
4. Check in UI - the first 10 transforms show 100%, the rest shows 0%

For example, this system contains transforms from farequote-airline-100 to farequote-airline-205. 
All transforms completed 100%.
All data frames contain 19 documents.
All were stopped using `POST _data_frame/transform/_all/_stop`.
However when querying using `GET _data_frame/transform/_all/_stats`, then the 10th transform shows checkpoint 0, stats 0 and has a missing progress object.

```
...
  {
      "id" : "farequote-airline-109",
      "state" : {
        "task_state" : "stopped",
        "indexer_state" : "stopped",
        "checkpoint" : 1,
        "progress" : {
          "total_docs" : 86274,
          "docs_remaining" : 0,
          "percent_complete" : 100.0
        }
      },
      "stats" : {
        "pages_processed" : 2,
        "documents_processed" : 86274,
        "documents_indexed" : 19,
        "trigger_count" : 1,
        "index_time_in_ms" : 7,
        "index_total" : 1,
        "index_failures" : 0,
        "search_time_in_ms" : 4,
        "search_total" : 2,
        "search_failures" : 0
      },
      "checkpointing" : {
        "operations_behind" : 0
      }
    },
    {
      "id" : "farequote-airline-110",
      "state" : {
        "task_state" : "stopped",
        "indexer_state" : "stopped",
        "checkpoint" : 0
      },
      "stats" : {
        "pages_processed" : 0,
        "documents_processed" : 0,
        "documents_indexed" : 0,
        "trigger_count" : 0,
        "index_time_in_ms" : 0,
        "index_total" : 0,
        "index_failures" : 0,
        "search_time_in_ms" : 0,
        "search_total" : 0,
        "search_failures" : 0
      },
      "checkpointing" : {
        "operations_behind" : 0
      }
    },
...
```

However when querying using `GET _data_frame/transform/farequote-airline-110/_stats` then the correct progress is returned.

```
{
  "count" : 1,
  "transforms" : [
    {
      "id" : "farequote-airline-110",
      "state" : {
        "task_state" : "stopped",
        "indexer_state" : "stopped",
        "checkpoint" : 1,
        "progress" : {
          "total_docs" : 86274,
          "docs_remaining" : 0,
          "percent_complete" : 100.0
        }
      },
      "stats" : {
        "pages_processed" : 2,
        "documents_processed" : 86274,
        "documents_indexed" : 19,
        "trigger_count" : 1,
        "index_time_in_ms" : 16,
        "index_total" : 1,
        "index_failures" : 0,
        "search_time_in_ms" : 14,
        "search_total" : 2,
        "search_failures" : 0
      },
      "checkpointing" : {
        "operations_behind" : 0
      }
    }
  ]
}
```

Note that if I repeat the test, but stop the transform by explicitly naming the transform id, then the `_all/_stats` returns the correct 100% responses.

Note that if I query `_all/_stats` at the point when all transforms are not yet stopped, then `_all/_stats` returns the correct 100% responses.

Weird ... but managed to repeat this.

### PR DESCRIPTION
**Pull request by davidkyle**
Set the size of the search request to the number of terms limited by 10,00. The default size of 10 is not appropriate.

Closes #43203

### CODE REVIEW

**Review by benwtrent — APPROVED** [review]

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java [hunk 1]
```diff
@@ -335,6 +335,7 @@ public void getTransformStats(Collection<String> transformIds, ActionListener<Li
         SearchRequest searchRequest = client.prepareSearch(DataFrameInternalIndex.INDEX_NAME)
                 .addSort(DataFrameField.ID.getPreferredName(), SortOrder.ASC)
                 .setQuery(builder)
+                .setSize(transformIds.size() > 10_000 ? 10_000 : transformIds.size())
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java]: [optional]
This can also be formulated as:
.setSize(Math.min(transformIds.size(), 10_000))

File: x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManagerTests.java [hunk 2]
```diff
@@ -239,7 +239,7 @@ public void testStateAndStats() throws InterruptedException {
     }
 
     public void testGetStateAndStatsMultiple() throws InterruptedException {
-        int numStats = randomInt(5);
+        int numStats = randomIntBetween(10, 15);
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManagerTests.java]: Is this change related to the one above? If so, how?

Comment (davidkyle) [reply x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManagerTests.java]: The default number of search results is 10. The bug is that the default value was used so only the first 10 documents are returned no matter how many stats documents there are. If this test was using a number > 10 in the first place the mistake would have been caught.

Comment (przemekwitek) [reply x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManagerTests.java]: Could you add a line comment here? Something like:
// Allow numStats greater than 10 so that...

**Review by przemekwitek — APPROVED** [review]
